File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.cpp
// #DEFINE MOD = 1000000007;
class Solution {
public:
    int numSub(string s) {
        long long c = 0, group = 0;
        for (auto i : s) {
            if (i == '1') {

                group++;
            } else {
                c += (group * (group + 1)) / 2;
                group=0;
            }
        }
        c += (group * (group + 1)) / 2;
        return c%1000000007;
    }
};

File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.py
class Solution:
    def numSub(self, s: str) -> int:
        c = 0
        group = 0
        for i in s:
            if i=='1':
                group+=1
            else:
                c+= (group*(group+1))//2
                group=0
        c+= (group*(group+1))//2
        return c%(10**9+7)
